File: 2379-maximum-total-importance-of-roads/2379-maximum-total-importance-of-roads.cpp
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        vector<long long> deg(n, 0);
        for(vector<int> &edge : roads){
            deg[edge[0]]++;
            deg[edge[1]]++;
        }
        sort(deg.begin(), deg.end());
        long long val = 1, ans = 0;
        for(long long d : deg){
            ans += (val * d);
            val++;
        }
        return ans;
    }
};

File: 2379-maximum-total-importance-of-roads/2379-maximum-total-importance-of-roads.java
class Solution {
    public long maximumImportance(int n, int[][] roads) {
        long[] deg = new long[n];
        for(int[] edge : roads){
            deg[edge[0]]++;
            deg[edge[1]]++;
        }
        Arrays.sort(deg);
        long val = 1, ans = 0;
        for(long d : deg){
            ans += (val*d);
            val++;
        }
        return ans;
    }
}